pkg/engine/renderer/enhancer.go

```diff
@@ -83,6 +69,106 @@ func (k *DefaultEnhancer) Apply(templates map[string]string, metadata Metadata)
 	return objs, nil
 }
 
+func addLabels(obj map[string]interface{}, metadata Metadata) error {
+	// List of paths for labels from here:
+	// https://github.com/kubernetes-sigs/kustomize/blob/master/api/konfig/builtinpluginconsts/commonlabels.go
+	labelPaths := [][]string{
```

Comment:
wouldn't it be better to "discover" the label locations?  this static approach works with existing kube objects...     We could "discover" the labels (granted it is by naming convention).

```
 k explain job --recursive | grep -A 10 labels
      labels	<map[string]string>
      managedFields	<[]Object>
         apiVersion	<string>
         fieldsType	<string>
         fieldsV1	<map[string]>
         manager	<string>
         operation	<string>
         time	<string>
      name	<string>
      namespace	<string>
      ownerReferences	<[]Object>
--
            labels	<map[string]string>
            managedFields	<[]Object>
               apiVersion	<string>
               fieldsType	<string>
               fieldsV1	<map[string]>
               manager	<string>
               operation	<string>
               time	<string>
            name	<string>
            namespace	<string>
            ownerReferences	<[]Object>
```

Reply:
My first approach was just to walk the yaml tree recursively, looking for metadata and applying the labels/annotations there. The problem with that is, that there's a lot of metadata that can't be changed once the resource is created. I think that's why Kustomize has a list of static places as well.